### PR TITLE
feat: add pocket network revenue + action

### PR DESCRIPTION
**Pull request by crisog**
This adds Pocket Network protocol revenue to Web3Index :rocket:

We need to provide a secret for the Github Action, called `POCKET_INFLUX_TOKEN` so please let us know so we can send it over.

### CODE REVIEW

File: registry.json [hunk 1]
```diff
@@ -1,84 +1,15 @@
...
+    "category": "Work Protocol",
+    "color": "#5289EA",
+    "everestID": "0x7ed21153c5c5eaa661af2821b6f90e838677c349",
+    "image": "https://web3index.org/logos/pocket.svg",
+    "name": "Pocket Network",
+    "stack": "Middleware",
+    "subcategory": "Infrastructure",
+    "symbol": "POKT",
+    "subgraph": false,
+    "genesisDate": "2020-07-28"
```
Comment (shreyaspapi) [inline registry.json:13]: We still need the other data here, why are the other protocols removed? @adamsoffer

File: public/logos/pocket.svg [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (adamsoffer) [inline public/logos/pocket.svg:1]: This is the wordmark logo. Needs to be replaced with Pocket's circular icon/badge version.

https://assets.website-files.com/609e7a6f2ec5c05d866ed6d3/60a7d6075adffd5ce48ea5da_POKT_Symbol_Color.svg

File: registry.json [hunk 3]
```diff
@@ -1,84 +1,15 @@
...
-    "stack": "Middleware",
-    "subcategory": "Compute",
-    "symbol": "AKT",
-    "usage": "https://akashlytics.com/web3-index/revenue",
-    "coingeckoID": "akash-network",
-    "genesisDate": "2020-09-24"
+  "pocket": {
+    "blockchain": "Pocket Network",
+    "category": "Work Protocol",
+    "color": "#5289EA",
```
Comment (adamsoffer) [inline registry.json:5]: "official" hex code appears to be `#1D8AED`

**Review by adamsoffer — CHANGES_REQUESTED** [review]

File: registry.json (hunk 1 above)
Comment (crisog) [reply registry.json:13]: > We still need the other data here, why are the other protocols removed? @adamsoffer

My bad. I removed them to test locally, and it sneaked into the PR.

Comment (crisog) [reply registry.json:95]: This has been reverted.

File: registry.json [hunk 4]
```diff
@@ -80,5 +80,17 @@
     "usage": "https://akashlytics.com/web3-index/revenue",
     "coingeckoID": "akash-network",
     "genesisDate": "2020-09-24"
+  },
+  "pocket": {
+    "blockchain": "Pocket Network",
```
Comment (adamsoffer) [inline registry.json:85]: Can we shorten to "Pocket"?

File: registry.json [hunk 5]
```diff
@@ -80,5 +80,17 @@
...
+    "category": "Work Protocol",
+    "color": "#1D8AED",
+    "everestID": "0x7ed21153c5c5eaa661af2821b6f90e838677c349",
+    "image": "https://web3index.org/logos/pocket.svg",
+    "name": "Pocket Network",
```
Comment (adamsoffer) [inline registry.json:90]: Can we shorten to "Pocket"?

File: cmd/pocket.ts [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+      name: name,
+    },
+  });
+
+  if (project == null) {
+    console.log(`Project ${name} doesn't exist. Create it`);
+    await prisma.project.create({
+      data: {
+        name: name,
+        lastImportedId: "1632268800",
```
Comment (adamsoffer) [inline cmd/pocket.ts:93]: On what date did Pocket protocol first start generating revenue? We need display all historical revenue but it's currently set to only display revenue after September 21, 2021 (1632268800).

**Review by adamsoffer — CHANGES_REQUESTED** [review]

File: cmd/pocket.ts (hunk 6 above)
Comment (crisog) [reply cmd/pocket.ts:93]: Historic revenue for Pocket is really tricky... as blockchains are enabled to generate revenue at different dates. It's not like blockchains are monetized from day 0, some take weeks to start being monetized or a few days.

Comment (crisog) [reply cmd/pocket.ts]: Finished the implementation for historical data :)

**Comment by crisog** [comment]
> @crisog can you please rebase with main?

It's good to go now.

**Review by adamsoffer — APPROVED** [review]

**Comment by adamsoffer** [comment]
Hey @crisog - looks like there are some price dates missing from your price endpoints which is causing the github action to fail. Can you ensure you have price data for all days? Thanks!

**Comment by adamsoffer** [comment]
@crisog Also, if you can provide more granular price feed data (hours or minutes as opposed to days) in the future that would be ideal. There can be non-trivial price swings within the span of 24 hours.

**Comment by crisog** [comment]
> Hey @crisog - looks like there are some price dates missing from your price endpoint which is causing the github action to fail. Can you ensure you have price data for all days? Thanks!

I see that price fetching fails for date `2021-05-28` and  `lastImportedId` is set to July 1st 2021. It shouldn't be fetching data prior to that.

**Comment by crisog** [comment]
> @crisog Also, if you can provide more granular price feed data (hours or minutes as opposed to days) in the future that would be ideal. There can be non-trivial price swings within the span of 24 hours.

This daily price is updated per hour as it was crafted for this use case, but we can make it more granular.

**Comment by adamsoffer** [comment]
> I see that price fetching fails for date 2021-05-28 and lastImportedId is set to July 1st 2021. It shouldn't be fetching data prior to that.

Okay that's helpful thanks. I'll start it from that date.

> This daily price is updated per hour as it was crafted for this use case, but we can make it more granular.

Yeah this would be useful if we have to reconstruct the price history from scratch again. Currently only daily prices are stored. ie http://ec2-35-177-209-25.eu-west-2.compute.amazonaws.com/prices/pokt?date_from=2021-05-27&date_to=2021-11-15

**Comment by crisog** [comment]
> > I see that price fetching fails for date 2021-05-28 and lastImportedId is set to July 1st 2021. It shouldn't be fetching data prior to that.
> 
> Okay that's helpful thanks. I'll start it from that date.
> 
> > This daily price is updated per hour as it was crafted for this use case, but we can make it more granular.
> 
> Yeah this would be useful if we have to reconstruct the price history from scratch again. Currently only daily prices are stored. ie http://ec2-35-177-209-25.eu-west-2.compute.amazonaws.com/prices/pokt?date_from=2021-05-27&date_to=2021-11-15

I will put this on our priority list for sure. Thanks for the feedback!

**Pull request merged**